lua/entities/gmod_wire_egp/lib/egplib/usefulfunctions.lua

```diff
@@ -442,3 +442,18 @@ function EGP:EGPCursor( this, ply )
 
 	return ReturnFailure( this )
 end
+
+function EGP.WorldToLocal(egp, object, x, y)
+	local _, realpos = EGP:GetGlobalPos(egp, object.index)
+	x, y = x - realpos.x, y - realpos.y
+	
+	local theta = math.rad(realpos.angle + object.angle or 0)
```

Comment:
I noticed this earlier and `realpos.angle + object.angle` seems wrong to me, unless GetGlobalPos doesn't incorporate object.angle itself. Might need to test and make sure this is working.

Reply:
It appears to be necessary, or I'm understanding something wrong. GetGlobalPos only seems to account for the parents' angles.

Reply:
Looking into it, it depends entirely on when `EGP:SetParent` is called. `object._angle` never gets modified on the server after this point (while it does, somehow, on the client). Cool feature.